### PR TITLE
feat: build portable binary by feature

### PR DESCRIPTION
**Pull request by zhangsoledad**
### What is changed and how it works?

Follow the same convention with rocksdb:

- Use best performance flags detected on the compiling machine by default.
- Build portable binary when the feature is specified.

### Check List 

Tests 

- Unit test
- Integration test

### Release note 

```release-note
Title Only: Include only the PR title in the release note.
```

### CODE REVIEW

**Comment by zhangsoledad** [comment]
https://github.com/nervosnetwork/rust-rocksdb/pull/37

File: db/Cargo.toml [hunk 1]
```diff
@@ -13,8 +13,12 @@ ckb-app-config = { path = "../util/app-config", version = "= 0.104.0-pre" }
 ckb-logger = { path = "../util/logger", version = "= 0.104.0-pre" }
 ckb-error = { path = "../error", version = "= 0.104.0-pre" }
 libc = "0.2"
-rocksdb = { package = "ckb-rocksdb", version ="=0.16.1", features = ["snappy"] }
+rocksdb = { package = "ckb-rocksdb", version ="=0.18.0", features = ["snappy"], default-features = false }
```
Comment (doitian) [inline db/Cargo.toml:16]: Please upgrade rocksdb in a separate PR.

Comment (zhangsoledad) [reply db/Cargo.toml:16]: https://github.com/nervosnetwork/ckb/pull/3468

**Comment by zhangsoledad** [comment]
https://github.com/nervosnetwork/ckb/pull/3460

**Review by quake — APPROVED** [review]

**Pull request closed without merging**